code/__DEFINES/misc_defines.dm

```diff
@@ -552,3 +552,10 @@
 #define BLOCKED_BURROWS "Blocked Burrows"
 #define CLASSIC_CAVES "Classic Caves"
 #define DEADLY_DEEPROCK "Deadly Deeprock"
+
+/// Takes a datum as input, returns its ref string, or a cached version of it
+/// This allows us to cache \ref creation, which ensures it'll only ever happen once per datum, saving string tree time
+/// It is slightly less optimal then a []'d datum, but the cost is massively outweighed by the potential savings
+/// It will only work for datums mind, for datum reasons
+/// : because of the embedded typecheck
+#define text_ref(datum) (isdatum(datum) ? (datum:cached_ref ||= "\ref[datum]") : ("\ref[datum]"))
```

Comment:
Should we use this in places we have existing `\ref` use?

Reply:
Should generally be used where it's guaranteed to be a `datum`. This will not work on non-datums. But yeah, generally, you can.

We have a ton of `\ref` useage still left.

515 improved the performance of `ref` significantly, so this may be a temporary thing, but this needs to be tested more thorough and verified.

Reply:
I mean we run 515 on live so